docs/src/implemented-proposals/tower-bft.md

```diff
@@ -90,42 +114,49 @@ The following parameters need to be tuned:
 - Number of votes in the stack before dequeue occurs \(32\).
 - Rate of growth for lockouts in the stack \(2x\).
 - Starting default lockout \(2\).
-- Threshold depth for minimum cluster commitment before committing to the fork \(8\).
+- Threshold check depth for minimum cluster commitment before committing to the fork \(8\).
 - Minimum cluster commitment size at threshold depth \(50%+\).
 
-### Free Choice
+### Fork Choice
 
-A "Free Choice" is an unenforcible validator action. There is no way for the protocol to encode and enforce these actions since each validator can modify the code and adjust the algorithm. A validator that maximizes self-reward over all possible futures should behave in such a way that the system is stable, and the local greedy choice should result in a greedy choice over all possible futures. A set of validator that are engaging in choices to disrupt the protocol should be bound by their stake weight to the denial of service. Two options exits for validator:
+Fork choice is how each validator determines which fork to vote on when multiple
+concurrent forks exist. Forks are weighted are based on the latest votes made by the validator set, and individual validators then vote on the "heaviest"
+such fork.
 
-- a validator can outrun previous validator in virtual generation and submit a concurrent fork
-- a validator can withhold a vote to observe multiple forks before voting
+Given the view of a single validator `i`:
 
-In both cases, the validator in the cluster have several forks to pick from concurrently, even though each fork represents a different height. In both cases it is impossible for the protocol to detect if the validator behavior is intentional or not.
+Let `V` be the set of "most recent" valid votes received by `i`, i.e., `v = (j, B)` is in `V` and if `i` has not also received a vote of the form `(j, B′) `such that `slot(B′) > slot(B)`.
 
-### Greedy Choice for Concurrent Forks
+Now the algorithm proceeds as follows:
 
-When evaluating multiple forks, each validator should use the following rules:
+1. For each vote `(j, B)` in `V`, add the stake of `j` to `B` and all of its
+ancestors.
+2. Now Set `B` to be the rooted block. Set `finish := 0`.
 
-1. Forks must satisfy the _Threshold_ rule.
-2. Pick the fork that maximizes the total cluster lockout time for all the ancestor forks.
-3. Pick the fork that has the greatest amount of cluster transaction fees.
-4. Pick the latest fork in terms of PoH.
+*While `finish == 0`
+*do*: If `i` has received no children of `B` then set `finish := 1` and return `B`. Otherwise let `B′` be the child of `B` (amongst those received by `i`) with most the most stake-weighted votes in `V`, breaking ties by the smallest slot.
```

Comment:
nit: as a statement this is too long, would be better to use
if (...)
else
...
to put it on multiple lines.

Reply:
Ah yeah, that's much better, thanks!